### PR TITLE
Issue #13389: Extend `getXdocsFilePaths` to gather `.xml.vm` files

### LINKED ISSUE #13389: Extend `getXdocsFilePaths` to gather `.xml.vm` files and get rid of `getAllXdocFilePaths`
In #13362 we add new method in `XdocUtil` - `getAllXdocFilePaths`. The method is almost identical to `getXdocsFilePaths` except that it also gets `.xml.vm` files. 

If we extend `getXdocsFilePaths` to include `.xml.vm` files, 3 tests break:
```diff
    public static Set<Path> getXdocsFilePaths() throws IOException {
        final Path directory = Paths.get(DIRECTORY_PATH);
        try (Stream<Path> stream = Files.find(directory, Integer.MAX_VALUE,
-                (path, attr) -> attr.isRegularFile() && path.toString().endsWith(".xml"))) {
+                (path, attr) -> {
+                   return attr.isRegularFile()
+                            && path.toString().endsWith(".xml")
+                            || path.toString().endsWith(".xml.vm");
+                })) {
            return stream.collect(Collectors.toSet());
        }
    }

```

```
[ERROR] Failures: 
[ERROR]   XdocsMobileWrapperTest.testAllCheckSectionMobileWrapper:67->iterateNode:80 writingjavadocchecks.xml.vm/HTML Code In Javadoc Comments: Tag 'table' in 'section' needs a wrapping <span> or <div> with the class 'wrapper'.
expected any of: [div, span]
but was        : section
[ERROR]   XdocsPagesTest.testAllSubSections:399 index.xml.vm sub-section Inactive / Old Tools for section Related Tools must match
expected: Related_Tools_Inactive_/_Old_Tools
but was : Related_Tools_Inactive_Old_Tools
[ERROR] Errors: 
[ERROR]   XdocsPagesTest.testAllXmlExamples:431->isValidCheckstyleXml:507 Checkstyle writingjavadocchecks.xml.vm has invalid Checkstyle xml (unable to parse configuration stream - Element type "audio" must be declared.:5:8): <audio><source src="horse.ogg" type="audio/ogg"/></audio>
[INFO] 
[ERROR] Tests run: 4202, Failures: 2, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE

```

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #13389

### CODE REVIEW

File: src/xdocs/index.xml.vm [hunk 1]
```diff
@@ -551,7 +551,7 @@
         </div>
       </subsection>
 
-      <subsection name="Inactive / Old Tools" id="Related_Tools_Inactive_Old_Tools">
+      <subsection name="Inactive / Old Tools" id="Related_Tools_Inactive_/_Old_Tools">
         <div class="wrapper">
```
Comment (romani) [inline src/xdocs/index.xml.vm:555]: Such id will not work, let's change `/`  to `or`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 2]
```diff
@@ -434,6 +434,7 @@ public void testAllXmlExamples() throws Exception {
                 // can't test ant structure, or old and outdated checks
                 assertWithMessage("Xml is invalid, old or has outdated structure")
                         .that(fileName.startsWith("anttask") || fileName.startsWith("releasenotes")
+                                || fileName.startsWith("writingjavadocchecks")
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:437]: Please put previous `||` on new line also to ease reading

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java [hunk 3]
```diff
@@ -60,7 +60,11 @@ private XdocUtil() {
     public static Set<Path> getXdocsFilePaths() throws IOException {
         final Path directory = Paths.get(DIRECTORY_PATH);
         try (Stream<Path> stream = Files.find(directory, Integer.MAX_VALUE,
-                (path, attr) -> attr.isRegularFile() && path.toString().endsWith(".xml"))) {
+                (path, attr) -> {
+                    return attr.isRegularFile()
+                            && path.toString().endsWith(".xml")
+                            || path.toString().endsWith(".xml.vm");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java:66]: Are you sure on this condition? You might mean `a && ( x || y)`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/index.xml.vm (hunk 1 above)
Comment (stoyanK7) [reply src/xdocs/index.xml.vm:555]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java (hunk 2 above)
Comment (stoyanK7) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:437]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java (hunk 3 above)
Comment (stoyanK7) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java]: Yes, done.

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]

**Comment by nrmancuso** [comment]
Failure in link check is not related to changes here.

**Pull request merged**